src/core/geometry/qgsrectangle.cpp

```diff
@@ -101,6 +101,15 @@ QgsRectangle &QgsRectangle::operator+=( const QgsVector v )
   return *this;
 }
 
+inline bool QgsRectangle::isNull() const
```

Comment:
My understanding is "inline" does nothing here, and this should remain in the header so that the compiler CAN optimise it.

Reply:
From what I read, the inline keyword on class member function definition acts they are defined inside the class header (for example: https://www.ibm.com/docs/en/i/7.5?topic=only-inline-member-functions-c). I have just test it successfully. I did not run any perf test (I do not know how to do it).

Reply:
Thanks!